process/publisher.go

```diff
@@ -0,0 +1,137 @@
+package process
+
+import (
+	"context"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-notifier-go/data"
+)
+
+type publisher struct {
+	handler PublisherHandler
+
+	broadcast                     chan data.BlockEvents
+	broadcastRevert               chan data.RevertBlock
+	broadcastFinalized            chan data.FinalizedBlock
+	broadcastTxs                  chan data.BlockTxs
+	broadcastBlockEventsWithOrder chan data.BlockEventsWithOrder
+	broadcastScrs                 chan data.BlockScrs
+
+	cancelFunc func()
+	closeChan  chan struct{}
+}
+
+// NewPublisher will create a new publisher component
+func NewPublisher(handler PublisherHandler) (*publisher, error) {
+	if check.IfNil(handler) {
+		return nil, ErrNilPublisherHandler
+	}
+
+	p := &publisher{
+		handler:                       handler,
+		broadcast:                     make(chan data.BlockEvents),
+		broadcastRevert:               make(chan data.RevertBlock),
+		broadcastFinalized:            make(chan data.FinalizedBlock),
+		broadcastTxs:                  make(chan data.BlockTxs),
+		broadcastScrs:                 make(chan data.BlockScrs),
+		broadcastBlockEventsWithOrder: make(chan data.BlockEventsWithOrder),
+		closeChan:                     make(chan struct{}),
+	}
+
+	return p, nil
+}
+
+// Run is launched as a goroutine and listens for events on the exposed channels
+func (p *publisher) Run() {
+	var ctx context.Context
+	ctx, p.cancelFunc = context.WithCancel(context.Background())
+
+	go p.run(ctx)
+}
+
+func (p *publisher) run(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			p.handler.Close()
+
+		case events := <-p.broadcast:
+			p.handler.Publish(events)
+		case revertBlock := <-p.broadcastRevert:
+			p.handler.PublishRevert(revertBlock)
+		case finalizedBlock := <-p.broadcastFinalized:
+			p.handler.PublishFinalized(finalizedBlock)
+		case blockTxs := <-p.broadcastTxs:
+			p.handler.PublishTxs(blockTxs)
+		case blockScrs := <-p.broadcastScrs:
+			p.handler.PublishScrs(blockScrs)
+		case blockEvents := <-p.broadcastBlockEventsWithOrder:
+			p.handler.PublishBlockEventsWithOrder(blockEvents)
+
+		}
+	}
+}
+
+// Broadcast will handle the block events pushed by producers and sends them to rabbitMQ channel
```

Comment:
send*

Reply:
removed that part of the comment since it's not related directly to rabbitmq anymore